corehq/apps/locations/models.py

```diff
@@ -242,10 +244,33 @@ class SQLLocation(MPTTModel):
     # This should really be the default location manager
     active_objects = OnlyUnarchivedLocationManager()
 
+    @classmethod
+    def _migration_get_fields(cls):
+        return ["domain", "name", "lineage", "site_code", "external_id",
+                "metadata", "is_archived", "latitude", "longitude"]
+
+    @classmethod
+    def _migration_get_couch_model_class(cls):
+        return Location
+
+    def _migration_do_sync(self):
+        couch_obj = self._migration_get_or_create_couch_object()
+        couch_obj._sql_location_type = self.location_type
+        self._migration_sync_to_couch(couch_obj)
+
     def save(self, *args, **kwargs):
         from corehq.apps.commtrack.models import sync_supply_point
         self.supply_point_id = sync_supply_point(self)
-        return super(SQLLocation, self).save(*args, **kwargs)
+
+        sync_to_couch = kwargs.pop('sync_to_couch', True)
+        kwargs['sync_to_couch'] = False  # call it here
+        super(SQLLocation, self).save(*args, **kwargs)
```

Comment:
The location_id is pulled from the couch location when it's [created](https://github.com/dimagi/dimagi-utils/blob/dc4bd18152475de87c91844a53c4d4ae4e700949/dimagi/utils/couch/migration.py#L158).  The couch doc with that id gets created on the initial save.  You can just pick an ID for new documents and send it over with the first save - couch doesn't need to create one for you.